### PR TITLE
[SourceSearch] Improve not-found UI

### PR DESCRIPTION
**Pull request by codehag**
Associated Issue: #3469 

### Summary of changes

* add absolute-center util
* remove flexbox
* update message
* add noNegativity option to allow sad face opt-out

Before:
screen shot 2017-07-27 at 12 25 14 pm

After:
screen shot 2017-07-27 at 12 24 05 pm

screen shot 2017-07-27 at 12 22 50 pm

Other places where search input is used still have the same styles:
screen shot 2017-07-27 at 12 22 42 pm

### CODE REVIEW

File: src/components/App.css [hunk 1]
```diff
@@ -37,3 +37,16 @@
...
+.absolute-center {
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  -webkit-transform: translateY(-50%);
+  -ms-transform: translateY(-50%);
+  transform: translateY(-50%);
+  -webkit-transform: translateX(-50%);
+  -ms-transform: translateX(-50%);
+  transform: translateX(-50%);
```
Comment (jasonLaster) [inline src/components/App.css]: autoprefixer + vertical-center could help

File: src/components/shared/SearchInput.js [hunk 2]
```diff
@@ -38,10 +39,13 @@ class SearchInput extends Component {
 
   static defaultProps: Object;
 
-  renderSvg() {
-    const { count, query } = this.props;
+  sadFaceStyles() {
```
Comment (jasonLaster) [inline src/components/shared/SearchInput.js]: perhaps `showSadFace`

File: src/components/shared/SearchInput.js [hunk 3]
```diff
@@ -128,7 +131,8 @@ class SearchInput extends Component {
 }
 
 SearchInput.defaultProps = {
-  size: ""
+  size: "",
+  noNegativity: false
```
Comment (jasonLaster) [inline src/components/shared/SearchInput.js]: `noNegativity: false` this is too hard on the brain

File: assets/panel/debugger.properties [hunk 4]
```diff
@@ -470,7 +470,7 @@ sourceSearch.search=Search Sources…
 
 # LOCALIZATION NOTE (sourceSearch.noResults): The center pane Source Search
 # message when the query did not match any of the sources.
-sourceSearch.noResults=No files matching %S found
```
Comment (jasonLaster) [inline assets/panel/debugger.properties]: this should be changed to `sourceSearch.noResults2`

File: src/components/shared/SearchInput.js [hunk 5]
```diff
@@ -38,10 +39,13 @@ class SearchInput extends Component {
 
   static defaultProps: Object;
 
-  renderSvg() {
-    const { count, query } = this.props;
+  hasErrorEmoji() {
```
Comment (jasonLaster) [inline src/components/shared/SearchInput.js]: perhaps, `shouldShowErrorEmoji`

File: src/components/shared/SearchInput.js [hunk 6]
```diff
@@ -128,7 +131,8 @@ class SearchInput extends Component {
 }
 
 SearchInput.defaultProps = {
-  size: ""
+  size: "",
+  showErrorEmoji: true
```
Comment (jasonLaster) [inline src/components/shared/SearchInput.js]: can you update the storybook stories with this new field?

File: assets/panel/debugger.properties (hunk 4 above)
Comment (codehag) [inline assets/panel/debugger.properties]: hmm, its not being used anywhere else -- why does this need to changed to have a new key?

**Comment by jasonLaster** [comment]
Found two things:

1. the label is not centered
2. the storybook does not show the sad face by default

**Comment by codehag** [comment]
So, we do not have autoprefixer on our project. If we were relying on it, it is likely other parts of our styles are broken. it exists on the launch pad but the launch pad doesn't have a roll in compiling our css. We need our own solution. For now I will add in the styles manually.

**Comment by codehag** [comment]
_or maybe i missed a comma... trying to check_

**Comment by codehag** [comment]
.._sigh_

**Pull request merged**